static/app/components/environmentPageFilter.tsx

```diff
@@ -0,0 +1,92 @@
+import {useState} from 'react';
+import {withRouter, WithRouterProps} from 'react-router';
+import styled from '@emotion/styled';
+
+import {updateEnvironments} from 'sentry/actionCreators/globalSelection';
+import DropdownButton from 'sentry/components/dropdownButton';
+import MultipleEnvironmentSelector from 'sentry/components/organizations/multipleEnvironmentSelector';
+import {IconWindow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import GlobalSelectionStore from 'sentry/stores/globalSelectionStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import space from 'sentry/styles/space';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+
+type Props = {
+  router: WithRouterProps['router'];
+  /**
+   * Reset these URL params when we fire actions
+   * (custom routing only)
+   */
+  resetParamsOnChange?: string[];
+};
+
+function EnvironmentPageFilter({router, resetParamsOnChange = []}: Props) {
+  const {projects, initiallyLoaded: projectsLoaded} = useProjects();
+  const organization = useOrganization();
+  const {selection, isReady} = useLegacyStore(GlobalSelectionStore);
+
+  const [selectedEnvironments, setSelectedEnvironments] = useState<string[] | null>(null);
+
+  const handleChangeEnvironments = (environments: string[] | null) => {
+    setSelectedEnvironments(environments);
+  };
+
+  const handleUpdateEnvironments = () => {
+    updateEnvironments(selectedEnvironments, router, {
+      save: true,
+      resetParams: resetParamsOnChange,
+    });
+    setSelectedEnvironments(null);
+  };
+
+  const customDropdownButton = ({isOpen, getActorProps, summary}) => {
+    return (
+      <StyledDropdownButton isOpen={isOpen} {...getActorProps()}>
+        <DropdownTitle>
+          <IconWindow />
+          {summary}
+        </DropdownTitle>
+      </StyledDropdownButton>
+    );
+  };
+
+  const customLoadingIndicator = (
+    <StyledDropdownButton showChevron={false} disabled>
+      <DropdownTitle>
+        <IconWindow />
+        {t('Loading\u2026')}
+      </DropdownTitle>
+    </StyledDropdownButton>
+  );
+
+  return (
+    <MultipleEnvironmentSelector
+      organization={organization}
+      projects={projects}
+      loadingProjects={!projectsLoaded || !isReady}
+      selectedProjects={selection.projects}
```

Comment:
Hmm, why does it need these 🤔

Reply:
i think just to determine which environments to show.

https://github.com/getsentry/sentry/blob/0bada19c8cb06f1f5e1c9607a84f2610bc30cf27/static/app/components/organizations/multipleEnvironmentSelector.tsx#L204